src/arrayMethodJoin.js

```diff
@@ -4,8 +4,43 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const length = this.length;
+    let firstElement = String(this[0]);
+    let separatorValue = separator;
+
+    if (length === 0) {
+      return '';
+    }
+
+    if (firstElement === 'undefined' || firstElement === 'null') {
+      firstElement = '';
+    }
+
+    if (length === 1) {
+      return firstElement;
+    }
+
+    if (separatorValue === undefined) {
+      separatorValue = ',';
+    }
+
+    if (separatorValue === null) {
+      separatorValue = 'null';
+    }
+
+    let result = firstElement;
```

Comment:
So let's set initial value to empty string

Reply:
fixed